### PR TITLE
Added ColorPicker Component

### PR DESCRIPTION
**Pull request by Eshno**
## Description

Added ColorPicker Component

Issue https://github.com/yisselda/SlackableThemes/issues/12

**In case the component causes any crash into the app, please comment the steps to reproduce.
I'm accepting any suggestions on this, so feel free to comment!**

## Testing Steps: 
1. Clone my fork `git clone https://github.com/Eshno/SlackableThemes/tree/colorpicker`
2. Checkout into the `colorpicker` branch `git checkout colorpicker`
3. Run `npm i`
4. Head over into the `PageContainer.js` file
5. Import the **ColorPicker Component** `import ColorPicker from '../components/ColorPicker/';`
6. Add up the following code (image and snippet below)
7. _Verify_ that the component updates the color properly on any change
8. _Verify_ that the component closes when clicked outside of it

**Color Picker Component**

Screen Shot 2019-10-06 at 10 09 24 AM

**Testing Code (lines marked in green and blue)**

Screen Shot 2019-10-06 at 9 57 09 AM

```
import React, {useState} from 'react'
import SideBar from '../components/SideBar';
import DashBoard from '../components/DashBoard';
import ColorPicker from '../components/ColorPicker/';

const PageContainer = props => {
  let [color, setColor] = useState('#00ffff');

  function updateColor(newColor) {
    setColor(newColor)
  }
  
  return (
    <div>
      <SideBar/>
      <DashBoard/>
      <span>{color}</span>
      <ColorPicker label="Color Picker" color={color} onChange={updateColor}/>
    </div>
    )
}
```

### CODE REVIEW

**Comment by Eshno** [comment]
Please Squash and Merge this PR when its ready to merge!

**Comment by yisselda** [comment]
@Eshno Thanks, I started trying out this component it looks good!
Can you add tests?

**Comment by Eshno** [comment]
Sure, I'll be working on that soon

File: src/tests/ColorPicker.test.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
 import React from 'react';
 import Enzyme from 'enzyme';
 import { shallow } from 'enzyme';
-import ColorPickers from '../components/ColorPickers';
```
Comment (yisselda) [inline src/tests/ColorPicker.test.js]: You need this line to import the component you are testing :)

Comment (Eshno) [reply src/tests/ColorPicker.test.js]: Thank you! I'm just currently trying to make the test work properly, having some issues with it 🙂

**Comment by Eshno** [comment]
@yisselda Did some small changes and added some tests on it, if anything misbehaves please tell me!

**Summary**:
- Added Tests
- Removed max length from the input (This was unnecessary)
- Small change on the Hex Validator (only verified the last digit in the string, now verifies the whole string)

**Pull request merged**